scripts/fdbshow

```diff
@@ -63,25 +63,51 @@ class FdbShow(object):
             return
 
         oid_pfx = len("oid:0x")
+
+        client = self.db.redis_clients["ASIC_DB"]
+        pipe = client.pipeline()
+        values = []
+        fdb_list = []
         for s in fdb_str:
             fdb_entry = s.decode()
             fdb = json.loads(fdb_entry .split(":", 2)[-1])
             if not fdb:
                 continue
-
-            ent = self.db.get_all('ASIC_DB', s, blocking=True)
+            fdb_list.append(fdb)
+            pipe.hgetall(s)
+        values = pipe.execute()
+        client = self.db.redis_clients["ASIC_DB"]
+        pipe = client.pipeline()
+        vlans = []
+        vlan_keys = self.db.keys('ASIC_DB', "ASIC_STATE:SAI_OBJECT_TYPE_VLAN:*")
+        for v in vlan_keys:
+            pipe.hgetall(v)
+        vlans = pipe.execute()
+
+        posi = 0
+        for ent in values:
+            if 'SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID' not in ent:
+                posi = posi + 1
+                continue
             br_port_id = ent[b"SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID"][oid_pfx:]
             ent_type = ent[b"SAI_FDB_ENTRY_ATTR_TYPE"]
             fdb_type = ['Dynamic','Static'][ent_type == "SAI_FDB_ENTRY_TYPE_STATIC"]
             if br_port_id not in self.if_br_oid_map:
+                posi = posi + 1
                 continue
             port_id = self.if_br_oid_map[br_port_id]
-            if_name = self.if_oid_map[port_id]
+            #When if_name not mapped to port_id in DB, using port_id as if_name in the display.
+            try:
+                if_name = self.if_oid_map[port_id]
+            except :
+                if_name = port_id
+
             if 'vlan' in fdb:
                 vlan_id = fdb["vlan"]
             elif 'bvid' in fdb:
-                vlan_id = port_util.get_vlan_id_from_bvid(self.db, fdb["bvid"])
-            self.bridge_mac_list.append((int(vlan_id),) + (fdb["mac"],) + (if_name,) + (fdb_type,))
+                vlan_id  = vlans[vlan_keys.index("ASIC_STATE:SAI_OBJECT_TYPE_VLAN:{}".format(fdb_list[posi][u'bvid']))] [b"SAI_VLAN_ATTR_VLAN_ID"]
```

Comment:
Do you see considerable improvement over using this vs `get_vlan_id_from_bvid`? Suggest to have this logic in one place.

Reply:
get_vlan_id_from_bvid reads from DB. But, we already have all the data retrieved from DB, so just need to parse that.

Reply:
I agree to that. My point is if `get_vlan_id_from_bvid` doesn't make much of a different in performance, suggest to use the API. Any change related to BRIDGE/VLAN/BVID in future (which happened before and we had to scan all the places where `vlan `id was used) will be contained in one place rather than in multiple places.

Reply:
understood. get_vlan_id_from_bvid is being used in 2 more places. I will see if I can change those caller also to retrieve all VLANs first and use it. Then I can modify get_vlan_id_from_bvid to not read from DB. Thanks.

Reply:
How much performance advantage are we getting without the call? Do you have any numbers?

Reply:
For 32K entries, it takes 7 seconds if I don't call get_vlan_id_from_bvid and it takes 5 min if I call get_vlan_id_from_bvid.

Reply:
oh ok, that's quite an improvement. Lets keep the change then!